src/shared-components/immersiveModal/index.tsx

```diff
@@ -22,193 +23,177 @@ import {
   ModalFooter,
   MainModalContentContainer,
 } from './style';
+import CrossIconComponent from './crossIconComponent';
 
 type ImmersiveModalProps = {
   children: React.ReactNode;
   headerImage?: React.ReactNode;
   footerContent?: React.ReactNode;
   onClose: () => void;
   title?: string;
+  [key: string]: unknown;
```

Comment:
`CHANGELOG`:

1. Expands `ImmersiveModalProps` since it appears we spread `...rest` into `<Overlay>`. 
1. Adds `FocusScope` to contain keyboard navigation inside the modal.
    - Inspiration drawn from `DialogModal` configuration, though this implementation required some additional hand-holding of the focus state (via `useFocusManager` in `CrossIconComponent`. The component has comments explaining the what/why). 
    - (The majority of the `ImmersiveModal` changes are, basically, kind of overkill to get the `contain` behavior working "optimally".)
1. Updates ID usage into `SCREAMING_CASE` variables, `getElementById` instead of `querySelector`, etc. 
1. Uses `useRef` as replacement for previous lifecycle method reliance to set instance variables.
1. Uses ES6 defaults in favor of `defaultProps`. 
1. Updates `CrossIconContainer` to be a `button` instead of a `div`. 
1. Uses `tabIndex` to remove the hidden button from keyboard navigation.